### PR TITLE
[Spark-13772][SQL] fix data type mismatch for decimal

### PR DESCRIPTION
**Pull request by cenyuhai**
fix data type mismatch for decimal, patch for branch 1.6

### CODE REVIEW

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/HiveTypeCoercionSuite.scala [hunk 1]
```diff
@@ -299,6 +299,19 @@ class HiveTypeCoercionSuite extends PlanTest {
     )
   }
 
+  test("test for SPARK-13772") {
+    val rule = HiveTypeCoercion.IfCoercion
+    ruleTest(rule,
+      If(Literal(true), Literal(1.0), Cast(Literal(1.0), DecimalType(19, 0))),
```
Comment (cloud-fan) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/HiveTypeCoercionSuite.scala]: Can you also check it with other database systems like Hive? Do they allow double type and decimal type in if and else branches?

Comment (cenyuhai) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/HiveTypeCoercionSuite.scala]: It is ok in hive 1.2.1 and spark 1.4.1. 
test case:
select if(1=1, cast(1 as double), cast(1 as decimal)) from test

**Comment by liancheng** [comment]
ok to test

**Comment by liancheng** [comment]
add to whitelist

**Comment by liancheng** [comment]
LGTM. Thanks for fixing this! Merging to branch-1.6.

**Comment by liancheng** [comment]
BTW, next time please use the format like `[SPARK-13772][SQL] <Title>` in PR title. For this one, I've manually updated the commit message while merging.

**Comment by dongjoon-hyun** [comment]
Hi, @cenyuhai .
If you don't mind, could you close this PR? It's already merged.
Currently, only PRs for master branch seems to close automatically.

**Pull request closed without merging**